lib/model/content.dart

```diff
@@ -504,6 +504,88 @@ class EmbedVideoNode extends BlockContentNode {
   }
 }
 
+class TableNode extends BlockContentNode {
+  const TableNode({super.debugHtmlNode, required this.rows});
+
+  final List<TableRowNode> rows;
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return rows
+      .mapIndexed((i, row) => row.toDiagnosticsNode(name: 'row $i'))
+      .toList();
+  }
+}
+
+class TableRowNode extends BlockContentNode {
+  const TableRowNode({
+    super.debugHtmlNode,
+    required this.cells,
+    required this.isHeader,
+  });
+
+  final List<TableCellNode> cells;
+
+  /// Indicates whether this row is the header row.
+  final bool isHeader;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<bool>('isHeader', isHeader));
+  }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return cells
+      .mapIndexed((i, cell) => cell.toDiagnosticsNode(name: 'cell $i'))
+      .toList();
+  }
+}
+
+/// The text-alignment that applies to text inside the table cell.
+///
+/// See GitHub-flavored Markdown:
+///   https://github.github.com/gfm/#tables-extension-
+// In Markdown, alignment is defined per column using the delimiter row.
+// However, the generated HTML specifies alignment for each cell in a row
+// individually, that matches the UI widget implementation which is also
+// row based and needs alignment information to be per cell.
+enum TableCellTextAlignment {
+  /// Text inside the table cell is left-aligned.
+  // Represented in markdown as `|: --- |` for the specific column in the delimeter row.
+  // TODO(i18n) RTL issues? https://github.com/zulip/zulip/issues/32265
+  left,
+  /// Text inside the table cell is center-aligned.
+  // Represented in markdown as `|: --- :|` for the specific column in the delimeter row.
+  center,
+  /// Text inside the table cell is right-aligned.
+  // Represented in markdown as `| --- :|` for the specific column in the delimeter row.
+  // TODO(i18n) RTL issues? https://github.com/zulip/zulip/issues/32265
+  right,
+  /// Text inside the table cell is aligned the default way.
+  // Represented in markdown as `| --- |` for the specific column in the delimeter row.
+  defaults
+}
```

Comment:
In how we name and document this enum, I'd like to be more explicit that what's being represented is a per-column setting, in the string from the server and in our model. That's how users experience the feature: we tell them they can set the alignment per-column, but not per individual cell (or for a row, or for the whole table, etc.).

It's true that the enum value is repeated on all of a column's cells, and our parsing logic has to understand that fact in order to recognize what the setting is for a given column. I also think it's fine for our model to repeat something on all of a column's cells too, because that's convenient for our parsing code and our UI code. But when doing those things, I'd still like the per-column abstraction to be clear somewhere, and I think the best place for that is in this enum's name and documentation. As in my proposal in https://github.com/zulip/zulip-flutter/pull/1031#discussion_r1831950371 :

```dart
// The text-alignment setting that applies to a cell's column, from the delimiter row.
//
// See GitHub-flavored Markdown:
//   https://github.github.com/gfm/#tables-extension-
enum TableColumnTextAlignment {
  /// All cells' text left-aligned, represented in Markdown as `|: --- |`.
  left, // TODO(i18n) RTL issues? https://github.com/zulip/zulip/issues/32265
  /// All cells' text center-aligned, represented in Markdown as `|: --- :|`.
  center,
  /// All cells' text right-aligned, represented in Markdown as `| --- :|`.
  right, // TODO(i18n) RTL issues? https://github.com/zulip/zulip/issues/32265
  /// Cells' text aligned the default way, represented in Markdown as `| --- |`.
  defaults,
}
```

In theory, our parsing code could grow a validation step to make sure the setting is expressed coherently, i.e., scan through a column's cells to check that their `style` attribute is consistent. I don't think we have to do that (could be expensive), but I do think it would be consistent with what's supposed to be represented.

I think the explicit per-column abstraction will be helpful when we come back to address the RTL issues. As part of that work, we'll want to get clear ideas of what we expect to happen vs. what's actually happening, at all of the steps from the compose-box input to how the message appears in the message list.